install/kubernetes/cilium/values.yaml.tmpl

```
@@ -622,6 +622,10 @@ gatewayAPI:
   # This will automatically set enable-envoy-config as well.
   enabled: false
 
+  # -- Metrics port for Gateway API controller
+  # If set to 0, metrics will be disabled
```

Nit: The periods will make a clearer separation when joining the lines for install/kubernetes/cilium/README.md and Documentation/helm-values.rst.

```suggestion
  # -- Metrics port for Gateway API controller.
  # If set to 0, metrics will be disabled.
```